src/main/java/com/minecolonies/coremod/commands/colonycommands/ColonyTeleportCommand.java

```diff
@@ -56,14 +73,73 @@ public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final En
         return colony.getPermissions().hasPermission(player, Action.TELEPORT_TO_COLONY);
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        if (null == colony)
+        {
+            final EntityPlayer player = actionMenu.getPlayerForArgument("player");
+            if (player != null)
+            {
+                IColony iColony = ColonyManager.getIColonyByOwner(server.getEntityWorld(), player);
+                if (null == iColony)
+                {
+                    if (sender instanceof EntityPlayer)
+                    {
+                        final Entity senderEntity = sender.getCommandSenderEntity();
+                        if (senderEntity != null)
+                        {
+                            final UUID mayorID = senderEntity.getUniqueID();
+                            if (iColony == null)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)